src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java

```diff
@@ -261,6 +266,14 @@ public Pod build() {
         jnlp.setEnv(new ArrayList<>(envVars.values()));
         if (jnlp.getResources() == null) {
             jnlp.setResources(new ContainerBuilder().editOrNewResources().addToRequests("cpu", new Quantity(DEFAULT_JNLP_CONTAINER_CPU_REQUEST)).addToRequests("memory", new Quantity(DEFAULT_JNLP_CONTAINER_MEMORY_REQUEST)).endResources().build().getResources());
+
+            if (DEFAULT_JNLP_CONTAINER_CPU_LIMIT!=null) {
+                jnlp.setResources(new ContainerBuilder().editOrNewResources().addToLimits("cpu", new Quantity(DEFAULT_JNLP_CONTAINER_CPU_LIMIT)).endResources().build().getResources());
```

Comment:
This is wrong. It is overriding the default request since you're starting from an empty ContainerBuilder.

Reply:
fix